scipy/signal/tests/test_signaltools.py

```diff
@@ -2507,6 +2507,108 @@ def test_bad_args(self):
 
 
 class TestPartialFractionExpansion(object):
+    def test_residue_general(self):
+        # Test are taken from issue #4464, note that poles in scipy are
+        # in increasing by absolute value order, opposite to MATLAB.
+        r, p, k = residue([5, 3, -2, 7], [-4, 0, 8, 3])
+        assert_almost_equal(r, [1.3320, -0.6653, -1.4167], decimal=4)
+        assert_almost_equal(p, [-0.4093, -1.1644, 1.5737], decimal=4)
+        assert_almost_equal(k, [-1.2500], decimal=4)
+
+        r, p, k = residue([-4, 8], [1, 6, 8])
+        assert_almost_equal(r, [8, -12])
+        assert_almost_equal(p, [-2, -4])
+        assert_equal(k.size, 0)
+
+        r, p, k = residue([4, 1], [1, -1, -2])
+        assert_almost_equal(r, [1, 3])
+        assert_almost_equal(p, [-1, 2])
+        assert_equal(k.size, 0)
+
+        r, p, k = residue([4, 3], [2, -3.4, 1.98, -0.406])
+        order = np.argsort(p)
+        assert_almost_equal(r[order],
+                            [-18.125 - 13.125j, -18.125 + 13.125j, 36.25])
+        assert_almost_equal(p[order], [0.5 - 0.2j, 0.5 + 0.2j, 0.7])
+        assert_equal(k.size, 0)
+
+        r, p, k = residue([2, 1], [1, 5, 8, 4])
+        assert_almost_equal(np.sort(r), [-1, 1, 3])
+        assert_almost_equal(np.sort(p), [-2, -2, -1])
+        assert_equal(k.size, 0)
+
+        r, p, k = residue([3, -1.1, 0.88, -2.396, 1.348],
+                          [1, -0.7, -0.14, 0.048])
+        assert_almost_equal(r, [-3, 4, 1])
+        assert_almost_equal(p, [0.2, -0.3, 0.8])
+        assert_almost_equal(k, [3, 1])
+
+        r, p, k = residue([1], [1, 2, -3])
+        assert_almost_equal(r, [0.25, -0.25])
+        assert_almost_equal(p, [1, -3])
+        assert_equal(k.size, 0)
+
+        r, p, k = residue([1, 0, -5], [1, 0, 0, 0, -1])
+        order = np.argsort(p)
+        assert_almost_equal(r[order], [1, 1.5j, -1.5j, -1])
+        assert_almost_equal(p[order], [-1, -1j, 1j, 1])
+        assert_equal(k.size, 0)
+
+        r, p, k = residue([3, 8, 6], [1, 3, 3, 1])
+        assert_almost_equal(np.sort(r), [1, 2, 3])
+        assert_allclose(p, -1)
+        assert_equal(k.size, 0)
+
+        r, p, k = residue([3, -1], [1, -3, 2])
+        assert_almost_equal(r, [-2, 5])
+        assert_almost_equal(p, [1, 2])
+        assert_equal(k.size, 0)
+
+        r, p, k = residue([2, 3, -1], [1, -3, 2])
+        assert_almost_equal(r, [-4, 13])
+        assert_almost_equal(p, [1, 2])
+        assert_almost_equal(k, [2])
+
+        r, p, k = residue([7, 2, 3, -1], [1, -3, 2])
+        assert_almost_equal(r, [-11, 69])
+        assert_almost_equal(p, [1, 2])
+        assert_almost_equal(k, [7, 23])
+
+        r, p, k = residue([2, 3, -1], [1, -3, 4, -2])
+        order = np.argsort(p)
+        assert_almost_equal(r[order], [4, -1 + 3.5j, -1 - 3.5j])
```

Comment:
32-bit failure is real:
```
        r, p, k = residue([2, 3, -1], [1, -3, 4, -2])
        order = np.argsort(p)
>       assert_almost_equal(r[order], [4, -1 + 3.5j, -1 - 3.5j])
E       AssertionError: 
E       Arrays are not almost equal to 7 decimals
E        ACTUAL: array([-1.+3.5j, -1.-3.5j,  4.+0.j ])
E        DESIRED: [4, (-1+3.5j), (-1-3.5j)]
```

Reply:
https://dev.azure.com/scipy-org/SciPy/_build/results?buildId=3600&view=logs&jobId=dca3794b-c3dc-52e6-709a-3545ea017448

Reply:
It happens when sorting close by real part roots. Now I've made the checking algorithm very robust (with `linear_sum_assignment`.).

Now the tests should pass reliably. Merging?